### PR TITLE
Add a SDKPatchTool

### PR DESCRIPTION
**Pull request by heng-liu**
This tool will download the latest dotnet from master branch, and patch it with the specified nupkgs.
It can be used for dotnet manual tests.

### CODE REVIEW

**Comment by heng-liu** [comment]
Thanks @nkolev92 ! I'll check if I can consolidate them.

**Comment by heng-liu** [comment]
Hi @nkolev92 , I found there're following differences between the two scripts:
1. The script you linked is getting dlls from artifacts\<project folder>, while this scrip is getting dlls from nupkgs. (as vendors can only see the artifacts/nupkgs folder)
2. The script you linked is patching the main 9 dlls, while this scrip is a full patching. So all dlls from nuget, and also the NuGet.targets, NuGet.props and so on will be patched. (so vendors can have a simulated patching to run dotnet command)
3. The script you linked is running on windows, while this scrip is for windows/Linux/Mac (some of the vendors manual test needs to be done on Mac/Linux, so this scrip will be used then)
Since this scrip is for vendors team to manual test dotnet related commands (donet list/dotnet restore), so I think those two scripts are of different uses.
Pls let me know if you have any suggestions. Thanks!

**Comment by nkolev92** [comment]
> The script you linked is getting dlls from artifacts<project folder>, while this scrip is getting dlls from nupkgs. (as vendors can only see the artifacts/nupkgs folder)

That's fair. I think there's a room for both, patching from artifacts and patching from the nupkgs. 
My suggestion was whether we should keep these scripts in the repo itself. It makes them easily available. 

> The script you linked is patching the main 9 dlls, while this scrip is a full patching. So all dlls from nuget, and also the NuGet.targets, NuGet.props and so on will be patched. (so vendors can have a simulated patching to run dotnet command)

I think the script does everything except NuGet props. It has a `NuGet.*`
https://github.com/NuGet/NuGet.Client/blob/2235bf9869c8fa4c7d441cc623f24cfd39bab621/scripts/nuget-debug-helpers.ps1#L144

I'll add NuGet.props to the script. 

> The script you linked is running on windows, while this scrip is for windows/Linux/Mac (some of the vendors manual test needs to be done on Mac/Linux, so this scrip will be used then)

That's fine my primary suggestion was to just make them available for everyone. 
Given that this is targeted for the vendors, I'm ok with this not being there. 

Personally I use the script in the in the repo and that's enough for me :) 

tldr; I'm ok with you merging this now.
